core/rfb.js

```diff
@@ -512,7 +502,9 @@ RFB.prototype = {
         // State change actions
 
         this._rfb_connection_state = state;
-        this.onupdatestate(this, state, oldstate);
+        var event = document.createEvent("CustomEvent");
+        event.initCustomEvent("updatestate", false, false, { state: state });
+        this.dispatchEvent(event);
```

Comment:
Can't we make a helper function for this?

Reply:
Possibly. But it was not that much code, so I figured it might be better to keep it obvious what is happening.

Reply:
fair enough